src/Consumer/ReferentMessageDispatcherConsumer.php

```diff
@@ -55,7 +53,10 @@ public function doExecute(array $data): bool
 
                 if ($i === MailjetService::PAYLOAD_MAXSIZE) {
                     $mailer->sendMessage(MailjetMessage::createFromModel($message, $chunk));
-                    $this->writeln($data['referent_uuid'], 'Message from '.$referent->getEmailAddress().' dispatched ('.$count.')');
+                    $this->writeln($data['uuid'], 'Message from '.$referent->getEmailAddress().' dispatched ('.$count.')');
+
+                    $referentMessage->incrementOffset($i);
+                    $manager->flush();
```

Comment:
You should flush after the loop.

Reply:
Actually, this whole PR is for this flush.
if the consumer fails during the loop, we need to know the "offset" value for the next try